### ISSUE TITLE
Retopoflow addon not working

### ISSUE THREAD

**Issue by ninjax-codes2004** [CGCookie/retopoflow#1252]
I keep on getting this problem after installation and when i am about to enable it (i am using blender 3.6 ):

File "C:\Users\myname\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\blenderregister.py", line 60, in <module>
    from . import retopoflow
  File "C:\Users\myname\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\retopoflow.py", line 37, in <module>
    from .rf.rf_sources         import RetopoFlow_Sources
  File "C:\Users\myname\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\rf\rf_sources.py", line 35, in <module>
    from ..rfmesh.rfmesh_render import RFMeshRender
  File "C:\Users\myname\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\rfmesh\rfmesh_render.py", line 43, in <module>
    from ...addon_common.common import bmesh_render as bmegl
  File "C:\Users\myname\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\addon_common\common\bmesh_render.py", line 105, in <module>
    verts_shader, verts_ubos = gpustate.gpu_shader('bmesh render: verts', verts_vs, verts_fs)
  File "C:\Users\myname\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\addon_common\common\gpustate.py", line 627, in gpu_shader
    shader = gpu.shader.create_from_info(shader_info)
Exception: Shader Compile Error, see console for more details

**Comment by imh1kiko** [comment]
Probably should also add basic info like "clean install or overwrite", and if restarted Blender if overwritten. Mine is picky with a lot of addons. There's also an alpha build somewhere in forums, so might be worth a shot?

**Comment by vxlcoder** [comment]
Thanks for the details, @ninjax-codes2004.

the last line (`Exception: Shader Compile Error, see console for more details`) says that something went wrong when trying to compile a shader, but the details on what went wrong are not reported here.

Could you please start Blender, toggle the system console, start RetopoFlow, then copy-paste _everything_ you see in the console here?  (see [help doc](https://docs.retopoflow.com/debugging.html) for more details)

This information will help us debug your issue and get you retopo'ing as quickly as possible!

**Comment by jaebird1** [comment]
Read prefs: "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\config\userpref.blend"
addon_utils.disable: RetopoFlow not disabled
Modules Installed (RetopoFlow) from 'C:\\Users\\jange\\Documents\\RetopoFlow_v3.4.0-GitHub.zip' into 'C:\\Users\\jange\\AppData\\Roaming\\Blender Foundation\\Blender\\3.6\\scripts\\addons'
Addon Common: use_bgl_default=False use_gpu_default=True use_gpu_scissor=True
Addon Common: {'backend': 'OPENGL', 'device': 'INTEL', 'vendor': 'Intel', 'renderer': 'Intel(R) HD Graphics 5500', 'version': '4.4.0 - Build 20.19.15.5058', 'compute_shader_support_get': True, 'max_batch_indices_get': 1048576, 'max_batch_vertices_get': 1048576, 'max_texture_layers_get': 2048, 'max_texture_size_get': 16384, 'max_textures_frag_get': 32, 'max_textures_geom_get': 32, 'max_textures_get': 192, 'max_textures_vert_get': 32, 'max_uniforms_frag_get': 4096, 'max_uniforms_vert_get': 4096, 'max_varying_floats_get': 64, 'max_vertex_attribs_get': 16, 'shader_image_load_store_support_get': True, 'shader_storage_buffer_objects_support_get': True}
RetopoFlow: Initial load
RetopoFlow git: None
RetopoFlow options path: C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\RetopoFlow_options.json
No options file
ERROR (gpu.shader): pyGPU_Shader VertShader:
      |
   10 | #extension GL_ARB_conservative_depth : enable
      |             ^
      | Error: 'interface_bmesh_render__verts' : two consecutive underscores are reserved for future use

ERROR (gpu.shader): pyGPU_Shader FragShader:
      |
   10 | #extension GL_ARB_conservative_depth : enable
      |    ^
      | Error: 'interface_bmesh_render__verts' : two consecutive underscores are reserved for future use

RetopoFlow: Unexpected Exception caught when trying to enable add-on!
Shader Compile Error, see console for more details
Traceback (most recent call last):
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\blenderregister.py", line 60, in <module>
    from . import retopoflow
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\retopoflow.py", line 37, in <module>
    from .rf.rf_sources         import RetopoFlow_Sources
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\rf\rf_sources.py", line 35, in <module>
    from ..rfmesh.rfmesh_render import RFMeshRender
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\rfmesh\rfmesh_render.py", line 43, in <module>
    from ...addon_common.common import bmesh_render as bmegl
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\addon_common\common\bmesh_render.py", line 105, in <module>
    verts_shader, verts_ubos = gpustate.gpu_shader('bmesh render: verts', verts_vs, verts_fs)
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\addon_common\common\gpustate.py", line 627, in gpu_shader
    shader = gpu.shader.create_from_info(shader_info)
Exception: Shader Compile Error, see console for more details

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\modules\addon_utils.py", line 333, in enable
    mod = __import__(module_name)
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\__init__.py", line 73, in <module>
    from .retopoflow import blenderregister
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\blenderregister.py", line 82, in <module>
    from .addon_common.common.debug import Debugger
ModuleNotFoundError: No module named 'RetopoFlow.retopoflow.addon_common'

Addon Common: DeepDebug should be initialized only once
RetopoFlow: Initial load
ERROR (gpu.shader): pyGPU_Shader VertShader:
      |
   10 | #extension GL_ARB_conservative_depth : enable
      |             ^
      | Error: 'interface_bmesh_render__verts' : two consecutive underscores are reserved for future use

ERROR (gpu.shader): pyGPU_Shader FragShader:
      |
   10 | #extension GL_ARB_conservative_depth : enable
      |    ^
      | Error: 'interface_bmesh_render__verts' : two consecutive underscores are reserved for future use

RetopoFlow: Unexpected Exception caught when trying to enable add-on!
Shader Compile Error, see console for more details
Traceback (most recent call last):
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\blenderregister.py", line 60, in <module>
    from . import retopoflow
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\retopoflow.py", line 37, in <module>
    from .rf.rf_sources         import RetopoFlow_Sources
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\rf\rf_sources.py", line 35, in <module>
    from ..rfmesh.rfmesh_render import RFMeshRender
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\rfmesh\rfmesh_render.py", line 43, in <module>
    from ...addon_common.common import bmesh_render as bmegl
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\addon_common\common\bmesh_render.py", line 105, in <module>
    verts_shader, verts_ubos = gpustate.gpu_shader('bmesh render: verts', verts_vs, verts_fs)
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\addon_common\common\gpustate.py", line 627, in gpu_shader
    shader = gpu.shader.create_from_info(shader_info)
Exception: Shader Compile Error, see console for more details

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\modules\addon_utils.py", line 333, in enable
    mod = __import__(module_name)
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\__init__.py", line 73, in <module>
    from .retopoflow import blenderregister
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\blenderregister.py", line 82, in <module>
    from .addon_common.common.debug import Debugger
ModuleNotFoundError: No module named 'RetopoFlow.retopoflow.addon_common'

Addon Common: DeepDebug should be initialized only once
RetopoFlow: Initial load
ERROR (gpu.shader): pyGPU_Shader VertShader:
      |
   10 | #extension GL_ARB_conservative_depth : enable
      |             ^
      | Error: 'interface_bmesh_render__verts' : two consecutive underscores are reserved for future use

ERROR (gpu.shader): pyGPU_Shader FragShader:
      |
   10 | #extension GL_ARB_conservative_depth : enable
      |    ^
      | Error: 'interface_bmesh_render__verts' : two consecutive underscores are reserved for future use

RetopoFlow: Unexpected Exception caught when trying to enable add-on!
Shader Compile Error, see console for more details
Traceback (most recent call last):
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\blenderregister.py", line 60, in <module>
    from . import retopoflow
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\retopoflow.py", line 37, in <module>
    from .rf.rf_sources         import RetopoFlow_Sources
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\rf\rf_sources.py", line 35, in <module>
    from ..rfmesh.rfmesh_render import RFMeshRender
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\rfmesh\rfmesh_render.py", line 43, in <module>
    from ...addon_common.common import bmesh_render as bmegl
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\addon_common\common\bmesh_render.py", line 105, in <module>
    verts_shader, verts_ubos = gpustate.gpu_shader('bmesh render: verts', verts_vs, verts_fs)
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\addon_common\common\gpustate.py", line 627, in gpu_shader
    shader = gpu.shader.create_from_info(shader_info)
Exception: Shader Compile Error, see console for more details

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\modules\addon_utils.py", line 333, in enable
    mod = __import__(module_name)
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\__init__.py", line 73, in <module>
    from .retopoflow import blenderregister
  File "C:\Users\jange\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\RetopoFlow\retopoflow\blenderregister.py", line 82, in <module>
    from .addon_common.common.debug import Debugger
ModuleNotFoundError: No module named 'RetopoFlow.retopoflow.addon_common'

**Comment by vxlcoder** [comment]
interesting... thanks for the helpful details, @jaebird1!

this looks like an issue with driver for Intel graphics, but will take a stab at it as soon as I can (hopefully this evening). I will need someone to test and provide feedback like this.

**Comment by vxlcoder** [comment]
@jaebird1, try installing v3.4.1 alpha from this temp link: https://www.dropbox.com/s/ycdml8zlwvxyqca/RetopoFlow_v3.4.1-alpha-GitHub.zip?dl=0

i _think_ it should fix the issue that you're getting.

**Comment by jaebird1** [comment]
Installing the alpha worked for me.  The GUI looks a little strange but the text is readable and the program seems to work perfectly. Thank you!

<img width="960" alt="retopolaunch1" src="https://github.com/CGCookie/retopoflow/assets/101355604/c298d1c6-982a-4e78-84f0-7270f2518d2d">
<img width="960" alt="retopoworking" src="https://github.com/CGCookie/retopoflow/assets/101355604/07959e6d-6054-4e31-a744-e074f4f3fd6c">

**Comment by imh1kiko** [comment]
Out of curiosity, could the error be thrown for when user has not selected GPU in System → Render Devices?

**Comment by vxlcoder** [comment]
@jaebird1, glad RF is at least working!  we used to have that visual bug issue with a past version of RF, and i thought we had fixed it, but i suppose not.  I will try to find that previous issue or create a new issue for this.

@imh1kiko, it shouldn't matter which render device is chosen, but please let me know if you see a difference.

@ninjax-codes2004, when you get a chance, please confirm whether installing RF v3.4.1 alpha (temp link: https://www.dropbox.com/s/ycdml8zlwvxyqca/RetopoFlow_v3.4.1-alpha-GitHub.zip?dl=0) fixes your original issue.  it seems to have worked for @jaebird1.

thanks everyone for helping debug this issue!

**Comment by gustavodelance30** [comment]
I'm having the same issue but i tried installing this one and still have the same error

**Comment by BUNZCANTMESH** [comment]
Hi my program stopped working

**Comment by jlampel** [comment]
It looks like the initial problem was solved, so I'll close this out. If RetopoFlow is not working for you in recent builds, please open a new issue!
